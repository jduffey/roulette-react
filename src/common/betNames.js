export const BET_NAMES = {
    STRAIGHT_UP_0: "StraightUp_0",
    STRAIGHT_UP_00: "StraightUp_00",
    STRAIGHT_UP_1: "StraightUp_1",
    STRAIGHT_UP_2: "StraightUp_2",
    STRAIGHT_UP_3: "StraightUp_3",
    STRAIGHT_UP_4: "StraightUp_4",
    STRAIGHT_UP_5: "StraightUp_5",
    STRAIGHT_UP_6: "StraightUp_6",
    STRAIGHT_UP_7: "StraightUp_7",
    STRAIGHT_UP_8: "StraightUp_8",
    STRAIGHT_UP_9: "StraightUp_9",
    STRAIGHT_UP_10: "StraightUp_10",
    STRAIGHT_UP_11: "StraightUp_11",
    STRAIGHT_UP_12: "StraightUp_12",
    STRAIGHT_UP_13: "StraightUp_13",
    STRAIGHT_UP_14: "StraightUp_14",
    STRAIGHT_UP_15: "StraightUp_15",
    STRAIGHT_UP_16: "StraightUp_16",
    STRAIGHT_UP_17: "StraightUp_17",
    STRAIGHT_UP_18: "StraightUp_18",
    STRAIGHT_UP_19: "StraightUp_19",
    STRAIGHT_UP_20: "StraightUp_20",
    STRAIGHT_UP_21: "StraightUp_21",
    STRAIGHT_UP_22: "StraightUp_22",
    STRAIGHT_UP_23: "StraightUp_23",
    STRAIGHT_UP_24: "StraightUp_24",
    STRAIGHT_UP_25: "StraightUp_25",
    STRAIGHT_UP_26: "StraightUp_26",
    STRAIGHT_UP_27: "StraightUp_27",
    STRAIGHT_UP_28: "StraightUp_28",
    STRAIGHT_UP_29: "StraightUp_29",
    STRAIGHT_UP_30: "StraightUp_30",
    STRAIGHT_UP_31: "StraightUp_31",
    STRAIGHT_UP_32: "StraightUp_32",
    STRAIGHT_UP_33: "StraightUp_33",
    STRAIGHT_UP_34: "StraightUp_34",
    STRAIGHT_UP_35: "StraightUp_35",
    STRAIGHT_UP_36: "StraightUp_36",
    FIRST_DOZEN: "1st 12",
    SECOND_DOZEN: "2nd 12",
    THIRD_DOZEN: "3rd 12",
    FIRST_18: "1 to 18",
    SECOND_18: "19 to 36",
    EVEN: "Even",
    ODD: "Odd",
    FIRST_COLUMN: "1st Column",
    SECOND_COLUMN: "2nd Column",
    THIRD_COLUMN: "3rd Column",
    RED: "Red",
    BLACK: "Black",
};
