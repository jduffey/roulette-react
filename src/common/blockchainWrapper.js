import { ethers } from "ethers";

const provider = new ethers.providers.JsonRpcProvider("http://localhost:8545");

// TODO: retrieve addresses from file generated by initialization script so they aren't hardcoded
const FIRST_PLAYER_ADDRESS = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266";
const SECOND_PLAYER_ADDRESS = "0x70997970C51812dc3A010C7d01b50e0d17dc79C8";
const THIRD_PLAYER_ADDRESS = "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC";
const HOUSE_ADDRESS = "0x90F79bf6EB2c4f870365E785982E1f101E93b906";
// Contract addresses are calculated from a hash of the deployer's address and nonce.
const TOKEN_CONTRACT_ADDRESS = "0x057ef64E23666F000b34aE31332854aCBd1c8544";
const ROULETTE_CONTRACT_ADDRESS = "0x261D8c5e9742e6f7f1076Fa1F560894524e19cad";

async function incrementTotalSpins() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const incrementTotalSpinsAbi = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function incrementTotalSpins()"],
        signer
    );
    const tx = await incrementTotalSpinsAbi.incrementTotalSpins();
    return tx;
}

async function getTotalSpins() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const getTotalSpinsAbi = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function getTotalSpins() public view returns (uint256)"],
        signer
    );
    const count = await getTotalSpinsAbi.getTotalSpins();
    return count;
}

async function getEthBalance(address) {
    const signer = provider.getSigner(address);
    const balance = await signer.getBalance();
    return ethers.utils.formatEther(balance);
}

// TODO leaving this for reference
// async function sendEther(provider, from, to, amount) {
//     const signer = provider.getSigner(from);
//     const tx = await signer.sendTransaction({
//         to: to,
//         value: ethers.utils.parseEther(amount)
//     });
//     return tx;
// }

async function getBlock() {
    const block = await provider.getBlock();
    return block;
}

async function getTokenBalance(address) {
    const signer = provider.getSigner(address);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function balanceOf(address) view returns (uint)"],
        signer
    );
    const balance = await token.balanceOf(address);
    const formattedBalance = ethers.utils.formatEther(balance);
    return formattedBalance;
}

async function depositEthForTokens(from, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function deposit() payable"],
        signer
    );
    const tx = await token.deposit({
        value: ethers.utils.parseEther(amount)
    });
    return tx;
}

async function redeemTokensForEth(from, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function withdraw(uint)"],
        signer
    );
    const tx = await token.withdraw(ethers.utils.parseEther(amount));
    return tx;
}

async function transferFrom(from, to, amount) {
    const signer = provider.getSigner(from);
    const token = new ethers.Contract(
        TOKEN_CONTRACT_ADDRESS,
        ["function transferFrom(address, address, uint) returns (bool)"],
        signer
    );
    const tx = await token.transferFrom(from, to, ethers.utils.parseEther(amount));
    return tx;
}

async function incrementJackpotBalance(amount) {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const contract = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function incrementJackpotBalance(uint256)"],
        signer
    );
    const tx = await contract.incrementJackpotBalance(ethers.utils.parseEther(amount));
    return tx;
}

async function getJackpotBalance() {
    const signer = provider.getSigner(HOUSE_ADDRESS);
    const contract = new ethers.Contract(
        ROULETTE_CONTRACT_ADDRESS,
        ["function getJackpotBalance() public view returns (uint256)"],
        signer
    );
    const balance = await contract.getJackpotBalance();
    const formattedBalance = ethers.utils.formatEther(balance);
    return formattedBalance;
}

let tokenSymbol;
(new ethers.Contract(
    TOKEN_CONTRACT_ADDRESS,
    ["function symbol() view returns (string)"],
    provider
)).symbol().then((symbol) => {
    tokenSymbol = symbol;
});

export {
    getEthBalance,
    getBlock,
    getTokenBalance,
    depositEthForTokens,
    redeemTokensForEth,
    transferFrom,
    incrementTotalSpins,
    getTotalSpins,
    incrementJackpotBalance,
    getJackpotBalance,
    FIRST_PLAYER_ADDRESS,
    SECOND_PLAYER_ADDRESS,
    THIRD_PLAYER_ADDRESS,
    HOUSE_ADDRESS,
    TOKEN_CONTRACT_ADDRESS,
    tokenSymbol,
};