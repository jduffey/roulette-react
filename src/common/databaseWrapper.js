const DATABASE_URL = new URL('/player', 'http://localhost:3001');

async function fetchPlayerBalance() {
    const res = await fetch(DATABASE_URL);
    if (!res.ok) {
        console.log("Not OK response from json-server: ", res);
        return `Error: ${res.status} ${res.statusText}`;
    }
    const json = await res.json();
    return json.balance;
}

async function updatePlayerBalance(balance) {
    const res = await fetch(DATABASE_URL, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify({ balance }),
    });
    if (!res.ok) {
        console.log("Not OK response from json-server: ", res);
    }
}

export { fetchPlayerBalance, updatePlayerBalance };
