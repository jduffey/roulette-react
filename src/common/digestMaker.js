// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
export async function digestMessage(message) {
    const msgUint8 = new TextEncoder().encode(message);

    const hashBuffer = await crypto.subtle.digest('SHA-256', msgUint8);

    const hashArray = Array.from(new Uint8Array(hashBuffer));

    const hashHex = hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');

    return hashHex;
}
