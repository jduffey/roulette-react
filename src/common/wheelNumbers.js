export const WHEEL_NUMBERS = {
    WN_0: "0",
    WN_00: "00",
    WN_1: "1",
    WN_2: "2",
    WN_3: "3",
    WN_4: "4",
    WN_5: "5",
    WN_6: "6",
    WN_7: "7",
    WN_8: "8",
    WN_9: "9",
    WN_10: "10",
    WN_11: "11",
    WN_12: "12",
    WN_13: "13",
    WN_14: "14",
    WN_15: "15",
    WN_16: "16",
    WN_17: "17",
    WN_18: "18",
    WN_19: "19",
    WN_20: "20",
    WN_21: "21",
    WN_22: "22",
    WN_23: "23",
    WN_24: "24",
    WN_25: "25",
    WN_26: "26",
    WN_27: "27",
    WN_28: "28",
    WN_29: "29",
    WN_30: "30",
    WN_31: "31",
    WN_32: "32",
    WN_33: "33",
    WN_34: "34",
    WN_35: "35",
    WN_36: "36",
}
