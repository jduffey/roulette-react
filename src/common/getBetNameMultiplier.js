import { BET_NAMES } from "./betNames";

export const getBetNameMultiplier = (betName) => {
    console.log("getBetNameMultiplier should be called once per bet placed", betName);
    switch (betName) {
        case BET_NAMES.STRAIGHT_UP_0:
        case BET_NAMES.STRAIGHT_UP_00:
        case BET_NAMES.STRAIGHT_UP_1:
        case BET_NAMES.STRAIGHT_UP_2:
        case BET_NAMES.STRAIGHT_UP_3:
        case BET_NAMES.STRAIGHT_UP_4:
        case BET_NAMES.STRAIGHT_UP_5:
        case BET_NAMES.STRAIGHT_UP_6:
        case BET_NAMES.STRAIGHT_UP_7:
        case BET_NAMES.STRAIGHT_UP_8:
        case BET_NAMES.STRAIGHT_UP_9:
        case BET_NAMES.STRAIGHT_UP_10:
        case BET_NAMES.STRAIGHT_UP_11:
        case BET_NAMES.STRAIGHT_UP_12:
        case BET_NAMES.STRAIGHT_UP_13:
        case BET_NAMES.STRAIGHT_UP_14:
        case BET_NAMES.STRAIGHT_UP_15:
        case BET_NAMES.STRAIGHT_UP_16:
        case BET_NAMES.STRAIGHT_UP_17:
        case BET_NAMES.STRAIGHT_UP_18:
        case BET_NAMES.STRAIGHT_UP_19:
        case BET_NAMES.STRAIGHT_UP_20:
        case BET_NAMES.STRAIGHT_UP_21:
        case BET_NAMES.STRAIGHT_UP_22:
        case BET_NAMES.STRAIGHT_UP_23:
        case BET_NAMES.STRAIGHT_UP_24:
        case BET_NAMES.STRAIGHT_UP_25:
        case BET_NAMES.STRAIGHT_UP_26:
        case BET_NAMES.STRAIGHT_UP_27:
        case BET_NAMES.STRAIGHT_UP_28:
        case BET_NAMES.STRAIGHT_UP_29:
        case BET_NAMES.STRAIGHT_UP_30:
        case BET_NAMES.STRAIGHT_UP_31:
        case BET_NAMES.STRAIGHT_UP_32:
        case BET_NAMES.STRAIGHT_UP_33:
        case BET_NAMES.STRAIGHT_UP_34:
        case BET_NAMES.STRAIGHT_UP_35:
        case BET_NAMES.STRAIGHT_UP_36:
            return 35;

        case BET_NAMES.FIRST_DOZEN:
        case BET_NAMES.SECOND_DOZEN:
        case BET_NAMES.THIRD_DOZEN:
        case BET_NAMES.FIRST_COLUMN:
        case BET_NAMES.SECOND_COLUMN:
        case BET_NAMES.THIRD_COLUMN:
            return 2;

        case BET_NAMES.FIRST_18:
        case BET_NAMES.SECOND_18:
        case BET_NAMES.EVEN:
        case BET_NAMES.ODD:
        case BET_NAMES.RED:
        case BET_NAMES.BLACK:
            return 1;

        default:
            throw new Error("Invalid bet name", betName);
    }
}
