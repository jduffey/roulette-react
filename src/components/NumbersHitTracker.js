import { WHEEL_NUMBERS } from "../common/wheelNumbers";

const CLASS_NAME = "NumbersHitTracker-component";
export function NumbersHitTracker(props) {
    const allHitNumbers = new Set();
    props.transactionHistory.forEach((tx) => {
        allHitNumbers.add(tx.spinResult);
    });

    return (
        <div
            className={CLASS_NAME}
        >
            {Object.values(WHEEL_NUMBERS).map((wheelNumber, i) => {

                const bgColor = allHitNumbers.has(wheelNumber) ?
                    "yellow" :
                    "inherit";
                const color = allHitNumbers.has(wheelNumber) ?
                    "black" :
                    "gray";
                return (
                    <div className="hit-number"
                        key={i}
                        style={{
                            backgroundColor: bgColor,
                            color: color,
                        }}
                    >
                        {wheelNumber}
                    </div>
                );
            })}
        </div>
    );
}
