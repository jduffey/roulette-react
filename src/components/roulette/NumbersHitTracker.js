import { useEffect, useState } from 'react';

import { WHEEL_NUMBERS } from "../../common/wheelNumbers";

import { getPlayerNumberCompletionSetCurrent } from "../../common/blockchainWrapper";

const CLASS_NAME = "NumbersHitTracker-component";
export function NumbersHitTracker(props) {
    const [currentSet, setCurrentSet] = useState(new Set());

    useEffect(() => {
        setTimeout(async () => {
            const currentNumbers = await getPlayerNumberCompletionSetCurrent(props.playerAddress);
            setCurrentSet(new Set(currentNumbers));
        }, 1000);
    }, [props.playerAddress, currentSet]);

    return (
        <div
            className={CLASS_NAME}
        >
            {Object.values(WHEEL_NUMBERS).map((stringyNumber, i) => {

                const wheelNumber = stringyNumber === "00"
                    ? 37
                    : parseInt(stringyNumber, 10);

                const backgroundColor = currentSet.has(wheelNumber) ?
                    "yellow" :
                    "inherit";
                const color = currentSet.has(wheelNumber) ?
                    "black" :
                    "gray";
                return (
                    <div className="hit-number"
                        key={i}
                        style={{
                            backgroundColor,
                            color,
                        }}
                    >
                        {wheelNumber === 37 ? "00" : wheelNumber}
                    </div>
                );
            })}
        </div>
    );
}
