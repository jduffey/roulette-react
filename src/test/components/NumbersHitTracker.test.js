import renderer from 'react-test-renderer';

import { WHEEL_NUMBERS } from '../../common/wheelNumbers';

import { NumbersHitTracker } from '../../components/NumbersHitTracker';

describe('NumbersHitTracker', () => {
    it('renders when there are no hit numbers', () => {
        const sut = NumbersHitTracker({
            transactionHistory: [],
        });

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });

    it('renders when there all numbers have been hit', () => {
        const sut = NumbersHitTracker({
            transactionHistory: Object.values(WHEEL_NUMBERS).map((wheelNumber) => {
                return {
                    "spinResult": wheelNumber,
                };
            }),
        });

        const actual = renderer.create(sut);

        expect(actual).toMatchSnapshot();
    });
});
