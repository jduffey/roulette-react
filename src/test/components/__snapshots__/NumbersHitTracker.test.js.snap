// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersHitTracker renders when there all numbers have been hit 1`] = `
<div
  className="NumbersHitTracker-component"
>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    0
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    00
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    1
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    2
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    3
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    4
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    5
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    6
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    7
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    8
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    9
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    10
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    11
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    12
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    13
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    14
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    15
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    16
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    17
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    18
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    19
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    20
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    21
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    22
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    23
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    24
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    25
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    26
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    27
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    28
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    29
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    30
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    31
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    32
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    33
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    34
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    35
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "yellow",
        "color": "black",
      }
    }
  >
    36
  </div>
</div>
`;

exports[`NumbersHitTracker renders when there are no hit numbers 1`] = `
<div
  className="NumbersHitTracker-component"
>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    0
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    00
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    1
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    2
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    3
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    4
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    5
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    6
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    7
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    8
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    9
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    10
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    11
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    12
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    13
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    14
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    15
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    16
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    17
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    18
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    19
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    20
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    21
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    22
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    23
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    24
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    25
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    26
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    27
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    28
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    29
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    30
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    31
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    32
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    33
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    34
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    35
  </div>
  <div
    className="hit-number"
    style={
      Object {
        "backgroundColor": "inherit",
        "color": "gray",
      }
    }
  >
    36
  </div>
</div>
`;
